src/createServer.js

```diff
@@ -1,3 +1,83 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const url = require('url');
+
+const { detectCase } = require('./convertToCase/detectCase');
+const { convertToCase } = require('./convertToCase/convertToCase');
+
+const createServer = () => {
+  const supportedCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+  const server = http.createServer((request, response) => {
+    response.setHeader('Content-Type', 'application/json');
+
+    const responseData = {};
+
+    responseData.errors = [];
+
+    const normalizedURL = new url.URL(
+      request.url,
+      `http://${request.headers.host}`,
+    );
+
+    const searchParams = normalizedURL.searchParams;
+
+    const textToConvert = normalizedURL.pathname.slice(1);
+    const targetCase = searchParams.get('toCase');
+
+    // If no 'text to convert', set error in errors array
+    if (!textToConvert.length) {
+      responseData.errors.push({
+        message: `Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".`,
+      });
+    }
+
+    // If no toCase query param
+    if (!searchParams.has('toCase')) {
+      responseData.errors.push({
+        message: `"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".`,
+      });
+
+      // If Case not supported -> Append error
+    } else if (targetCase === null || !supportedCases.includes(targetCase)) {
+      responseData.errors.push({
+        message: `This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.`,
```

Comment:
for better maintainability instead of hardcoded "SNAKE, KEBAB, CAMEL, PASCAL, UPPER." in string you should use supportedCases array. This ensures that the list of valid cases is consistent and easier to update